src/main/java/org/mule/extension/http/internal/request/HttpPollingSourceUtils.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.extension.http.internal.request;
+
+import static java.lang.String.format;
+import static java.util.Optional.empty;
+import static org.mule.extension.http.internal.request.HttpPollingSource.ATTRIBUTES_PLACEGHOLDER;
+import static org.mule.extension.http.internal.request.HttpPollingSource.ITEM_PLACEGHOLDER;
+import static org.mule.extension.http.internal.request.HttpPollingSource.PAYLOAD_PLACEGHOLDER;
+import static org.mule.extension.http.internal.request.HttpPollingSource.WATERMARK_PLACEGHOLDER;
+import static org.mule.runtime.api.i18n.I18nMessageFactory.createStaticMessage;
+import static org.mule.runtime.api.metadata.DataType.STRING;
+import static org.mule.runtime.api.metadata.MediaType.APPLICATION_JAVA;
+
+import org.mule.extension.http.api.HttpResponseAttributes;
+import org.mule.runtime.api.el.BindingContext;
+import org.mule.runtime.api.el.ExpressionLanguage;
+import org.mule.runtime.api.exception.MuleRuntimeException;
+import org.mule.runtime.api.metadata.MediaType;
+import org.mule.runtime.api.metadata.TypedValue;
+import org.mule.runtime.extension.api.runtime.operation.Result;
+
+import java.io.Serializable;
+import java.util.Optional;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+
+/**
+ * Utils to be used for resolution of expressions of the {@link HttpPollingSource}
+ * @since 1.7
+ */
+public final class HttpPollingSourceUtils {
+
+  private HttpPollingSourceUtils() {
+    // Empty private constructor to avoid instantiation.
+  }
+
+  private static BindingContext buildContext(Optional<TypedValue<String>> payload, Optional<HttpResponseAttributes> attributes,
+                                             Serializable currentWatermark, Optional<TypedValue> item) {
+    BindingContext.Builder builder = BindingContext.builder();
+    payload.ifPresent(p -> builder.addBinding(PAYLOAD_PLACEGHOLDER, p));
+    attributes.ifPresent(attr -> builder.addBinding(ATTRIBUTES_PLACEGHOLDER, TypedValue.of(attr)));
+    builder.addBinding(WATERMARK_PLACEGHOLDER, TypedValue.of(currentWatermark));
+    item.ifPresent(it -> builder.addBinding(ITEM_PLACEGHOLDER, it));
+    return builder.build();
+  }
+
+  private static Result<TypedValue<?>, HttpResponseAttributes> toResult(TypedValue<?> item, MediaType mediaType,
+                                                                        HttpResponseAttributes attributes) {
+    return Result.<TypedValue<?>, HttpResponseAttributes>builder().attributes(attributes).output(item).mediaType(mediaType)
+        .build();
+  }
+
+  private static boolean isJavaPayload(MediaType mediaType) {
+    return mediaType.equals(APPLICATION_JAVA.withCharset(mediaType.getCharset().orElse(null)));
```

Comment:
use `matches` instead of equals. That not only allows you not not worry about the charset, but also about other potential additional properties.